integration-tests/e2e_tests_ci_tool/cmd/override_test_config_cmd.go

```diff
@@ -0,0 +1,207 @@
+package cmd
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"github.com/pelletier/go-toml/v2"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	ctf_config "github.com/smartcontractkit/chainlink-testing-framework/config"
+)
+
+const DryRunFlag = "dry-run"
+
+// fromBase64TOML is a base64 encoded TOML config to override
+var fromBase64TOML string
+
+var overrideTestConfigCmd = &cobra.Command{
+	Use:   "override",
+	Short: "Override base64 encoded TOML config with provided flags. Overrides only existing fields in the base config.",
+	Run: func(cmd *cobra.Command, _ []string) {
+		dryRun, _ := cmd.Flags().GetBool(DryRunFlag)
+
+		decoded, err := base64.StdEncoding.DecodeString(fromBase64TOML)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error decoding base64 TOML: %v\n", err)
+			os.Exit(1)
+		}
+		var baseConfig ctf_config.TestConfig
+		err = toml.Unmarshal(decoded, &baseConfig)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error unmarshalling base64 TOML: %v\n", err)
+			os.Exit(1)
+		}
+
+		cmd.Flags().Visit(func(flag *pflag.Flag) {
```

Comment:
I still believe that we don't need any explicit logging about overrides and it's better to print the whole config when secrets are properly migrated (and they'll be masked).
Overriding things explicitly still need maintenance when you add a new variable, in several places.
Marshaling to the same pointer is the easiest way IMO.
Either we read `base64` from CI, marshal default config, marshal overrides again on the same pointer and prints the result, or if we run locally we marshal the default config and then `overrides.toml`.
This way there will be no additional code to change when we add a new param + we print all the stuff, secrets masked in CI, if you put something that is not in our vault it's not a secret.

Reply:
> it's better to print the whole config when secrets are properly migrated

Totally! I'd really like to print it out to make it visible how default config was overridden. I even have a code to do it that is commented until we remove secrets from `TestConfig` struct https://github.com/smartcontractkit/chainlink/blob/TT-1297-Create-nightly-pipeline-for-all-e2e-tests-in-core/.github/actions/create-default-e2e-config-override/action.yml#L142-L144

Reply:
@skudasov also, please note that this go tool as well as .github/actions/create-default-e2e-config-override/action.yml GHA replaces the existing bash scripts that we had to create test config override. It's intermediary step before we have 1pass support on CI. And, a big improvement over the current bash scripts since we can unit test the golang tool I created

Reply:
hmmm... what's the purpose of using this tool? IMHO if we really want it, we should make that part of `TestConfig`, since there are zillions levels of overrides.